### ISSUE TITLE
mp4-hls.py: ERROR: failed to write samples (-23)

### ISSUE THREAD

**Issue by bcharna** [axiomatic-systems/Bento4#133]
I create 5 videos using mp4-dash-encode.py. Then I use these 5 videos as the input to mp4-hls.py. This normally works without any issue. However, there seems to be a video for which calling mp4-hls.py with it results in "ERROR: failed to write samples (-23)". Looking at a little C++ source I see -23 means AP4_ERROR_NOT_ENOUGH_SPACE. Do you have any recommendations to resolve this? the same source video can go into a DASH manifest via mp4-dash.py with no problem and normally I don't have this issue with hls manifest generation.

**Comment by barbibulle** [comment]
This error code would seem to indicate that there isn't enough space on disk to save the output file. Maybe the file is too large?
If that's not the issue, is it possible to provide access to the file that causes this problem so that I could reproduce the issue?

**Comment by DamianImrich** [comment]
I was facing same issue after trying to package three qualities to hls. When I tried package these files separately I figured out that **"ERROR: failed to write samples (-23)"** was only problem with biggest quality(1080p) or second biggest(720p) with bigger videos. Some smaller videos was packaged successfully with all qualities in it. 

All these qualities were ffmpeg converted videos from source video to different resolutions and bitrates. All qualities were fragmented by **mp4fragment** tool to package it to DASH firstly.

I thought it must be problem with mp4fragment tool. So I tried to refragment problematic quality (1080p) with **ffmpeg -movflags faststart**.

`ffmpeg -i 1080p_mp4fragment.mp4 -c copy -movflags faststart 1080p_ffmpegfragment.mp4`

It was completed with great speed because **-c copy argument** tells ffmpeg to not reencode it.

So I tried package all three qualities to hls with ffmpeg fragmented 1080p and other qualities mp4fragmented and it started to working!!

Packaging was successful after refragmenting with ffmpeg!
